markdown runner can't find unnamed class's main

The markdown runner can't invoke unnamed classes.
````
```java
void main() {
  System.out.println("Hello world!");
}
```
````

it says it can't find the main class.
```
$ jbang README.md 
[jbang] [ERROR] No main class deduced, specified nor found in a manifest nor jar
[jbang] Run with --verbose or -x for more details. The --verbose or -x must be placed before the jbang command. I.e. jbang --verbose run [...]
```
```
 jbang -x README.md 
[jbang] [ERROR] No main class deduced, specified nor found in a manifest nor jar
dev.jbang.cli.ExitException: No main class deduced, specified nor found in a manifest nor jar
	at dev.jbang.source.generators.JarCmdGenerator.generateCommandLineList(JarCmdGenerator.java:271)
	at dev.jbang.source.generators.BaseCmdGenerator.generate(BaseCmdGenerator.java:51)
	at dev.jbang.cli.Run.doCall(Run.java:110)
	at dev.jbang.cli.BaseCommand.execute(BaseCommand.java:134)
	at org.aesh.command.impl.operator.EndOperator.execute(EndOperator.java:41)
	at org.aesh.command.impl.Executions$ExecutionImpl.execute(Executions.java:239)
	at org.aesh.command.impl.AeshCommandRuntime.runExecutor(AeshCommandRuntime.java:222)
	at org.aesh.command.impl.AeshCommandRuntime.executeCommand(AeshCommandRuntime.java:175)
	at org.aesh.AeshRuntimeRunner.execute(AeshRuntimeRunner.java:160)
	at dev.jbang.Main.main(Main.java:46)
```

A workaround is to just put the statements.
````
```java
System.out.println("Hello world!");
```
````

--- a/src/main/java/dev/jbang/source/sources/MarkdownSource.java
+++ b/src/main/java/dev/jbang/source/sources/MarkdownSource.java
@@ -12,6 +12,9 @@ import dev.jbang.cli.BaseCommand;
 import dev.jbang.cli.ExitException;
 import dev.jbang.resources.ResourceRef;
 import dev.jbang.resources.resolvers.LiteralScriptResourceResolver;
+import dev.jbang.source.BuildContext;
+import dev.jbang.source.Builder;
+import dev.jbang.source.CmdGeneratorBuilder;
 import dev.jbang.source.Source;
 import dev.jbang.util.Util;
 
@@ -24,6 +27,11 @@ public class MarkdownSource extends JshSource {
 	@Override
 	public @NonNull Type getType() {
 		return Type.markdown;
+	}
+
+	@Override
+	public Builder<CmdGeneratorBuilder> getBuilder(BuildContext ctx) {
+		return ctx.getProject().isJShell() ? super.getBuilder(ctx) : new JavaAppBuilder(ctx);
 	}
 
 	public static Source create(ResourceRef resourceRef, Function<String, String> replaceProperties) {

--- a/src/test/java/dev/jbang/cli/TestRun.java
+++ b/src/test/java/dev/jbang/cli/TestRun.java
@@ -241,6 +241,34 @@ public class TestRun extends BaseTest {
 	}
 
 	@Test
+	void testMarkdownWithMainMethod() throws IOException {
+		Path readme = jbangTempDir.resolve("readme.md");
+		writeString(readme,
+				"```java\nclass Readme {\n    public static void main(String... args) {\n        System.out.println(\"Hello\");\n    }\n}\n```\n");
+
+		Run run = JBang.parseCommand("run", readme.toString());
+		Project prj = run.createProjectBuilderForRun().build(readme.toString());
+		CmdGeneratorBuilder genb = Project.codeBuilder(BuildContext.forProject(prj)).build();
+
+		String result = run.updateGeneratorForRun(genb).build().generate();
+
+		assertThat(result, matchesPattern("^.*java(.exe(\\^\\\")?)? .*$"));
+		assertThat(result, containsString("Readme"));
+	}
+
+	@Test
+	void testMarkdownWithUnnamedMainMethod() throws Exception {
+		assumeTrue(JavaUtil.getCurrentMajorJavaVersion() >= 25);
+		Path readme = jbangTempDir.resolve("readme.md");
+		writeString(readme,
+				"```java\nvoid main() {\n    System.out.println(\"Hello unnamed\");\n}\n```\n");
+
+		CaptureResult<Integer> result = checkedRun(readme.toString());
+
+		assertThat(result.out, containsString("Hello unnamed"));
+	}
+
+	@Test
 	void testMarkdownWindows() throws IOException {
 		Path readmeFileOrg = examplesTestFolder.resolve("readme.md");
 		String readmeText = Util.readString(readmeFileOrg);

